src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -0,0 +1,302 @@
+import com.intellij.openapi.roots.ProjectRootManager;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.plugins.haxe.lang.lexer.HaxeTokenTypes;
+import com.intellij.plugins.haxe.lang.psi.*;
+import com.intellij.plugins.haxe.util.HaxeClassModel;
+import com.intellij.plugins.haxe.util.HaxeMethodModel;
+import com.intellij.plugins.haxe.util.HaxePsiUtils;
+import com.intellij.plugins.haxe.util.HaxeResolveUtil;
+import com.intellij.psi.*;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.psi.util.PsiUtil;
+import com.intellij.util.IncorrectOperationException;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+public class HaxeSemanticAnnotator implements Annotator {
+
+
+  @Override
+  public void annotate(PsiElement element, AnnotationHolder holder) {
+    PsiFile file = element.getContainingFile();
+    // Analyze upon file changes
+    Long analyzedFileTime = file.getUserData(HaxeSemanticError.KEY_ANALYZED_METHOD_TIME);
+    if (analyzedFileTime == null || analyzedFileTime != file.getModificationStamp()) {
+      analyze(file);
+      file.putUserData(HaxeSemanticError.KEY_ANALYZED_METHOD_TIME, file.getModificationStamp());
+    }
+
+    List<HaxeSemanticError> errors = element.getUserData(HaxeSemanticError.KEY_HAXE_ERROR);
+    if (errors != null) {
+      for (HaxeSemanticError error : errors) {
+        Annotation annotation = holder.createErrorAnnotation(element, error.message);
+        if (error.quickfix != null) annotation.registerFix(error.quickfix);
+      }
+    }
+  }
+
+
+  static void analyze(final PsiElement element) {
+    HaxeSemanticError.clearErrors(element);
+
+    if (element instanceof PsiJavaToken) {
+      if (((PsiJavaToken)element).getTokenType() == HaxeTokenTypes.LITINT) {
+        //element.putUserData(KEY_HAXE_ERROR, "litint!");
+      }
+    }
+    else if (element instanceof HaxePackageStatement) {
+      PackageChecker.check((HaxePackageStatement)element);
+    }
+    else if (element instanceof HaxeMethod) {
+      HaxeClass clazz = HaxePsiUtils.getAncestor(element, HaxeClass.class);
+      final HaxeDeclarationAttribute overrideAttribute = HaxePsiUtils.getChild(element, HaxeDeclarationAttribute.class, "override");
+      HaxeClassModel parentClass = clazz.getModel().getExtendingClass();
+      boolean requiredOverride = false;
+
+      if (parentClass != null) {
+        HaxeMethodModel method = parentClass.getMethod(((HaxeMethod)element).getModel().getName());
+        if (method != null) {
+          requiredOverride = true;
+        }
+      }
+
+      //System.out.println(aClass);
+      if (overrideAttribute != null && !requiredOverride) {
+        HaxeSemanticError.addError(
```

Comment:
Nope. They are mandatory. Indeed I lost lot of time, when I changed an interface to an abstract class in haxe because IDE was not telling me that I needed the override, so I detected that in compile time. Not sure if that's the case of Java, but most languages enforce its usage.

```
$ haxe Test.hx
Test.hx:2: lines 2-3 : Field a should be declared with 'override' since it is inherited from superclass Test2
```

```
class Test extends Test2 {
        public function a() {
        }
        static public function main() {
        }
}

class Test2 {
        public function a() {
        }
}
```